core/src/main/java/org/apache/accumulo/core/client/ClientSideIteratorScanner.java

```diff
@@ -134,15 +142,22 @@ public SamplerConfiguration getSamplerConfiguration() {
       return samplerConfig;
     }
 
-    @Deprecated(since = "2.1.0")
     @Override
-    public ServiceEnvironment getServiceEnv() {
-      return new ClientServiceEnvironmentImpl(context.get());
+    @Deprecated(since = "2.0.0")
+    public AccumuloConfiguration getConfig() {
+      var ctx = context.get();
+      try {
+        return new ConfigurationCopy(
+            ctx.tableOperations().getConfiguration(ctx.getTableName(tableId.get())));
+      } catch (AccumuloException | TableNotFoundException e) {
+        throw new RuntimeException("Error getting table configuration", e);
+      }
     }
 
+    @Deprecated(since = "2.1.0")
     @Override
-    public PluginEnvironment getPluginEnv() {
-      return new ClientServiceEnvironmentImpl(context.get());
```

Comment:
This is the method that should be implemented. The `getServiceEnv()` method only existed for backwards compatibility. The code should not rely on the interface implementation of `getPluginEnv()` calling the `getServiceEnv()` here, because that behavior is removed when the deprecated code is dropped. Instead, it's best to keep both methods, as before your change, so that when the deprecated one is dropped in subsequent versions, the correct implementation is kept.

Reply:
Thanks, this was something I was unsure about when implementing this. Some `IteratorEnvironment`s had both `getServiceEnv()` and `getPluginEnv()` (e.g., `OfflineIterator`) while some had just `getServiceEnv()` (e.g., `TabletIteratorEnvironment`), and I wasn't sure which to go with. I'll fix the impls to use both.